tembo-operator/src/app_service/ingress.rs

```diff
@@ -269,7 +269,7 @@ pub fn generate_ingress_tcp_routes(
                             continue;
                         }
 
-                        let matcher = format!("{host_matcher} && PathPrefix(`{}`)", path);
+                        let matcher = format!("HostSNI(`{}`)", host_matcher);
```

Comment:
You may need to update some assertions in the `functional_test_app_service` integration test